src/main/java/reactor/kafka/receiver/internals/DefaultKafkaReceiver.java

```diff
@@ -322,11 +322,20 @@ private void start() {
     private void fail(Throwable e, boolean async) {
         log.error("Consumer flux exception", e);
         recordSubmission.error(e);
-        cancel(async);
+        dispose(async);
     }
 
-    private void cancel(boolean async) {
-        log.debug("cancel {}", isActive);
+    private void dispose() {
+        boolean isEventsThread = Thread.currentThread()
```

Comment:
That might be too simplistic - what about keeping the reference in a ThreadLocal that can be populated on start by N schedule or similar.

Reply:
fixed